src/shaders/gaussianFilter.frag

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// This is an implementation based on:
+// https://rastergrid.com/blog/2010/09/efficient-gaussian-blur-with-linear-sampling/
+
+precision highp float;
+
+// ------------ input -----------------------
+in highp vec2 textureCoordinates;
+
+// ------------ uniforms --------------------
+uniform highp sampler2D SourceTexture;
+uniform bool FilterHorizontally;  // true on x or false on y axis
+
+//------------- output ----------------------
+layout(location = OUTPUT_ATTRIBUTE_LOCATION_COLOR) out highp vec4 fragmentColor;
+
+//------------- shader ----------------------
+//
+void main(void) {
+  ivec2 dims = textureSize(SourceTexture, 0);  // lod = 0
+  float scale = 1.0 / dims.x;
+  if (!FilterHorizontally) {
+    scale = 1.0 / dims.y;
+  }
```

Comment:
Continuing from the previous suggestion, with `vec2 filterDirection` this would be 

```suggestion
  vec2 scale2 = filterDirection/dims;
  float scale = scale2.x + scale2.y; // or max(), one of those is always 0
```

Reply:
Eh, actually, instead of using `texture()` and calculating `scale`, you should be using `texelFetch()` that takes integer coordinates. Since that skips filtering, it could potentially be also faster. A "level up" from there is `texelFetchOffset()`, since you have a compile-time offset here it's a great match. 

Not sure about NV in particular, probably won't affect perf much, but the reason for the `*Offset()` variant was to allow the hardware to predict which area of the texture will be accessed by a shader invocation so it can prefetch the memory. I guess it could have some non-zero impact on mobile hardware (VR, etc).

- - -

Possible further optimization idea -- when I was implementing gaussian blur about a decade ago for a silly game project, I got the best perf by [unrolling the loop by hand](https://github.com/nuclearwhales/push-the-box/blob/248540eec4b052dc210716647994e364d3672c33/src/Shaders/Blur.frag#L20-L37). Because the thing had to run on NaCl (a predecessor to WebGL 1), I couldn't use `texelFetch()` nor `texelFetchOffset()` and had to instead pass the texture coordinates from the vertex shader to avoid the HW use slow dependent texture reads. Since you can use `texelFetch()`, that part makes no sense anymore, but I still think unrolling may help ... or maybe `texelFetchOffset()` will force you to do that anyway, since it expects the offset to be a constant expression.